### DISCUSSION TITLE
ERT2 & ERT3: conflicts, harmony and strangulation

### DISCUSSION THREAD

**Discussion by markusdregi** [equinor/ert#2229] [General]
**Context**
ERT2 is built on a rather hard assumption of a shared disk that can be used for data communication between the client machine and the compute cluster, while ERT3 is intended to use other means for transporting this data. In particular, the intention is that ERT3 will be built around the following assumptions:

- no shared disk with read and write access used for data communication between client and compute,
- a central storage for data and
- an explicit definition of the compute environment.

**Approach**
The [development strategy](https://ert.readthedocs.io/en/latest/architecture/dev-strategy.html) states that we should use the strangler pattern to hone out responsibilities related to analysis, ensemble evaluation and data storage. However, it might be worth considering whether the business logic driving the experiments (the run models) also should be encapsulated and strangulated in a similar fashion.

**Comment by markusdregi** [comment]
Some related discussions in: https://github.com/equinor/ert/issues/2170

**Comment by pinkwah** [comment]
At the very least we should rename ERT3 to ERT-next or something along those lines. As it currently stands, we have put ourselves in a position where bumping the major version of ERT2 is icky.
